doc: Update CONTRIBUTING.md (#562)
canonical/mongodb-operator: CONTRIBUTING.md

This document explains the processes and practices recommended for contributing enhancements to this operator.

> **Note:** The charm's python business logic is written in a shared library that can be found [here](https://github.com/canonical/mongo-single-kernel-library). This where python contributions should be made. 

Reviewer: ```suggestion
**Note:** The charm's python business logic is written in a shared library that can be found [here](https://github.com/canonical/mongo-single-kernel-library). This is where python contributions should be made. 
```

This document explains the processes and practices recommended for contributing enhancements to this operator.

> Please note that all the logic of this operator is stored in a shared library hosted [here](https://github.com/canonical/mongo-single-kernel-library).

Reviewer: ```suggestion
**Note:** The charm's python business logic is written in a shared library that can be found [here](https://github.com/canonical/mongo-single-kernel-library). This where python contributions should be made. 
```

- Generally, before developing enhancements to this charm, you should consider opening an issue [on Single Kernel repository](https://github.com/canonical/mongo-single-kernel-library/issues) explaining your use case.

> - If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).

Reviewer: ```suggestion
- If you would like to chat with us about your use-cases or proposed implementation, you can reach us on our [Matrix channel](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or in [Discourse](https://discourse.charmhub.io/).
```

- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
- All enhancements require review before being merged. Additionally, new code must pass the tests. Code review typically examines
  - code quality
  - test coverage
  - user experience for Juju administrators of this charm.
- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto the `main` branch. This also avoids merge commits and creates a linear Git commit history.

> - Once the code has been merged on the Single Kernel repository, wait for a new version of the python package to be published, and create a PR on this repository that bumps the version of the package.

Reviewer: ```suggestion
- Once the code has been merged on the [repository](https://github.com/canonical/mongo-single-kernel-library/) of the Mongo Single Kernel lib, wait for a new version of the [python package](https://pypi.org/project/mongo-charms-single-kernel/) to be published, and create a PR on this repository that bumps the version of the package.
```

> - If you added some new interfaces, please don't forget to add them here.

Reviewer: I would also before that they should open bump dependency PRs in all 4 repos. We shouldn't allow bumping in a single repo.

juju deploy ./mongodb_ubuntu-20.04-amd64.charm
```

## Intended use case

> TODO: expand on this section when spec doc is created

Reviewer: Let's remove
